### ISSUE TITLE
Upgrade `ttx`

### ISSUE THREAD

**Issue by timvandermeij** [mozilla/pdf.js#11802]
The current version of `ttx` that we use is over five years old; see https://github.com/fonttools/fonttools/tree/8388a2e37ce349dac6555bb824c82723e3b65fbf (the current submodule commit). We should really update to the latest release, not only to be up-to-date but also because our version of `ttx` requires Python 2 which has been discouraged for a long time now and is (finally) out of support.

This issue has come up locally when I ran `gulp fonttest` and two tests failed. It turns out our script spawns `python` to run `ttx`, but `python` on modern Linux systems refers to Python 3. For Python 2, the `python2` command must be used. When I changed this, the tests passed.

This will require that we have Python 3 on the bots, which is also highly recommended nowadays and might already be there.

**Comment by timvandermeij** [comment]
@brendandahl Since you upgraded Node.js on the bots, do you also happen to know if we have Python 3 on the bots and which minor version (3.7, 3.8, ...)? Once we can run Python 3 on the bots, we can look into this upgrade.

**Comment by timvandermeij** [comment]
I looked into how we could move forward here since using an ancient version of `ttx`doesn't feel good: the current version 2.5 is from September 2014, which is almost 8 years old now. The good news is that we can update to a newer version, namely 3.19.0 from November 2017, but the bad news is that we can't update any further than that. The reasons for this are:
    
- From version 4.0.0 onward `ttx` is a Python 3-only application, which is completely understandable since Python 2 is officially out of support now and already discouraged for a long time. However, our current infrastructure (bots) still run old OS versions that don't have (modern enough) Python 3 versions, so for the time being we're stuck with Python 2 unfortunately. Note that the infrastructure issue is tracked in #11851.

- From version 3.20.0 onward the upstream project included a custom `unicodedata` module that shadows the standard library module, which is not great in general, but problematic for us since it leads to import errors due to the way we call `ttx`. We use a Git submodule to fetch the code and then run `ttx` using a custom invocation with `PYTHONPATH` set, however, this causes the library path to be first in `sys.path` and therefore cause the shadowed version to override the standard version; see https://docs.python.org/3/tutorial/modules.html#the-module-search-path. Normally in Python one would not use Git submodules and `PYTHONPATH` at all, but instead create a virtual environment, install `fonttools` using `pip` and simply use the provided `ttx` entrypoint (see https://github.com/fonttools/fonttools/blob/main/setup.py#L467) so one doesn't have to know about internal details like the location of the `ttx.py`file. Technically we could already do this now, but for Python 2 this requires the separate `virtualenv` package to be installed, whereas in Python 3 it's built-in which makes it much easier to work with. Therefore, we should also only do this once we're running on Python 3.

It would also be better to not use regexes to check the `ttx` output, but since it's XML just use an XML parser, which would make future updates easier and makes the code easier to read and less fragile overall.

In short, the following points can be defined:

- [ ] Update `ttx` from version 2.5 to version 3.19.0.
  - [ ] Fix the `test/font/ttxdriver.js` file to call Python with `python2`.
  - [ ] Fix the `test/ttx/README.md` file to drop the obsolete note about 32-bit Python 2.6.
- [ ] Use XML parsing instead of regexes in the font tests.
- [ ] Get #11851 done to unblock further work here.
- [ ] Replace Git submodule approach with a proper virtual environment.
- [ ] Update `ttx` from version 3.19.0 to the most recent version (Python 3-only).
